src/hotspot/cpu/aarch64/aarch64_neon.ad

```diff
@@ -5263,9 +5263,14 @@ instruct vsra8B_imm(vecD dst, vecD src, immI shift) %{
   format %{ "sshr    $dst, $src, $shift\t# vector (8B)" %}
   ins_encode %{
     int sh = (int)$shift$$constant;
-    if (sh >= 8) sh = 7;
-    __ sshr(as_FloatRegister($dst$$reg), __ T8B,
-           as_FloatRegister($src$$reg), sh);
+    if (sh == 0) {
+      __ orr(as_FloatRegister($dst$$reg), __ T8B,
+             as_FloatRegister($src$$reg), as_FloatRegister($src$$reg));
+    } else {
+      if (sh >= 8) sh = 7;
+      __ sshr(as_FloatRegister($dst$$reg), __ T8B,
```

Comment:
I think we should add an assert to make sure 0 is not passed to the assembler.

Reply:
Agree, I'll do this.